src/cpp/src/speculative_decoding/speculative_decoding_stateful_eagle3.cpp

```diff
@@ -0,0 +1,1242 @@
+                  std::to_string(total_len) + ")");
+}
+
+InferenceOutput Eagle3DraftModelWrapper::infer(const ov::Tensor& input_ids,
+                                               const ov::Tensor& attention_mask,
+                                               const ov::Tensor& position_ids,
+                                               const ov::Tensor& hidden_states) {
+    const size_t input_token_count = input_ids.get_shape()[1];
+
+    log_debug(LogCategory::ITER, "Draft inference: " + std::to_string(input_token_count) + " tokens");
+    log_model_inputs(input_ids, attention_mask, position_ids);
+
+    // Set standard inputs
+    m_request.set_tensor("input_ids", input_ids);
+    m_request.set_tensor("attention_mask", attention_mask);
+    m_request.set_tensor("position_ids", position_ids);
+
+    // Set hidden states (either from target model or internal)
+    OPENVINO_ASSERT(hidden_states && hidden_states.get_size() > 0, "hidden_states must be provided");
+    auto shape = hidden_states.get_shape();
+    OPENVINO_ASSERT(shape.size() == 3, "Invalid hidden states shape: ", format_shape(shape));
+
+    log_debug(LogCategory::ITER, "Using hidden states: " + format_shape(shape));
+    m_request.set_tensor("hidden_states", hidden_states);
+
+    if (m_device != "NPU") {
+        m_request.get_tensor("beam_idx").set_shape({BATCH_SIZE});
+        m_request.get_tensor("beam_idx").data<int32_t>()[0] = 0;
+    }
+
+    // Execute inference
+    uint64_t time_us = execute_inference();
+    update_performance_metrics(time_us, input_token_count);
+
+    // Collect outputs
+    InferenceOutput output;
+    output.logits = get_logits();
+    output.hidden_features = get_hidden_features();
+
+    log_model_outputs(output.logits, output.hidden_features);
+    log_debug(LogCategory::ITER, "Draft inference done: " + std::to_string(time_us / 1000.0) + "ms");
+
+    return output;
+}
+
+InferResult Eagle3DraftModelWrapper::forward(const InferContext& ctx) {
+    log_debug(LogCategory::ITER,
+              "Draft forward: input_token_count=" + std::to_string(ctx.input_token_count) +
+                  ", use_target_hidden=" + std::to_string(ctx.use_target_hidden));
+
+    // 1. Prepare inputs
+    ov::Tensor input_ids, attention_mask, position_ids;
+    build_model_inputs(ctx.input_token_count, input_ids, attention_mask, position_ids);
+
+    // 2. Get hidden states from appropriate source
+    ov::Tensor hidden_states;
+    if (ctx.use_target_hidden) {
+        OPENVINO_ASSERT(ctx.source_sequence, "source_sequence required when use_target_hidden=true");
+        hidden_states = ctx.source_sequence->get_hidden_state();
+        OPENVINO_ASSERT(hidden_states && hidden_states.get_size() > 0, "Source sequence has no hidden state");
+    } else {
+        hidden_states = get_current_sequence()->get_hidden_state();
+        OPENVINO_ASSERT(hidden_states && hidden_states.get_size() > 0, "Own sequence has no hidden state");
```

Comment:
Could you please rewrite message to `Own sequence contains invalid hidden state`